Stratis.Bitcoin/Logging/Logs.cs

```diff
@@ -8,56 +8,149 @@ namespace Stratis.Bitcoin.Logging
 {
 	public class Logs
 	{
-		static Logs()
+		public static void Configure(ILoggerFactory factory)
 		{
-			Configure(new LoggerFactory());
+			LoggerFactory = factory;
+
+			// These match namespace; classes can also use CreateLogger<T>, which will inherit
+			Configuration = factory.CreateLogger("Stratis.Bitcoin.Configuration");
+			RPC = factory.CreateLogger("Stratis.Bitcoin.RPC");
+			FullNode = factory.CreateLogger("Stratis.Bitcoin.FullNode");
+			ConnectionManager = factory.CreateLogger("Stratis.Bitcoin.Connection");
+			Bench = factory.CreateLogger("Stratis.Bitcoin.FullNode.ConsensusStats");
+			Mempool = factory.CreateLogger("Stratis.Bitcoin.MemoryPool");
+			BlockStore = factory.CreateLogger("Stratis.Bitcoin.BlockStore");
+			EstimateFee = factory.CreateLogger("Stratis.Bitcoin.Fee");
 		}
 
-		public static void Configure(ILoggerFactory factory)
+		public static ILoggerFactory GetLoggerFactory(string[] args)
 		{
-			Configuration = factory.CreateLogger("Configuration");
-			RPC = factory.CreateLogger("RPC");
-			FullNode = factory.CreateLogger("FullNode");
-			ConnectionManager = factory.CreateLogger("ConnectionManager");
-			Bench = factory.CreateLogger("Bench");
-			Mempool = factory.CreateLogger("Mempool");
-			BlockStore = factory.CreateLogger("BlockStore");
-			EstimateFee = factory.CreateLogger("EstimateFee");
+			// TODO: preload enough args for -conf= or -datadir= to get debug args from there
+
+			//Configuration = factory.CreateLogger("Configuration");
+			//FullNode = factory.CreateLogger("FullNode");
+			//ConnectionManager = factory.CreateLogger("ConnectionManager");
+			//EstimateFee = factory.CreateLogger("EstimateFee");
+
+			var debugArgs = args.Where(a => a.StartsWith("-debug=")).Select(a => a.Substring("-debug=".Length).Replace("\"", "")).FirstOrDefault();
```

Comment:
NodeArgs makes several calls to logging, so need to configure logging *before* we get full NodeArgs (similar to how the beginning of NodeArgs manually pulls out data directory, config file, and network).

If you move it to after NodeArgs, then you would have to remove the logging from NodeArgs (or do it differently).

Generally the earlier the logging is configured the better, so yes, that means outside the normal configuration.